### DISCUSSION TITLE
add a custom wallet and checked for conditions

### DISCUSSION THREAD

**Discussion by rajputsandeep** [rainbow-me/rainbowkit#1880] [Q&A]
I have to add infinity wallet to rainbowkit custom wallet and I am not able to add infinity wallet to us tb I have to call openInfinityWallet function, is this possible with custom wallet please help me 
![infinity](https://github.com/rainbow-me/rainbowkit/assets/39721765/28b45074-5093-4c4b-9407-430118fdd603)
![infinity2](https://github.com/rainbow-me/rainbowkit/assets/39721765/2f1e544f-2f6b-4606-950c-a5ca8e83a3cf)

**Comment by rajputsandeep** [comment]
anyone please help me for this

**Reply by NORAVemp** [reply]
Can you test if you can call your custom wallet like this?

**Comment by 0xmkh** [comment]
@rajputsandeep Can you add `hidden` property to your custom wallet ? Here is an example: https://github.com/rainbow-me/rainbowkit/blob/main/packages/rainbowkit/src/wallets/walletConnectors/safeWallet/safeWallet.ts#L11-L14
